lib/authorization.rb

```diff
@@ -66,24 +66,16 @@ def self.included(base)
       end
     end
 
-    def hyper_tag_identifier
-      :name
-    end
-
-    def hyper_tag_prefix
-      self.class.name
-    end
-
-    def hyper_tag_name
-      "#{hyper_tag_prefix}/#{send(hyper_tag_identifier)}"
+    def hyper_tag_name(project = nil)
+      raise NoMethodError
     end
 
     def hyper_tag(project)
       AccessTag.where(project_id: project.id, hyper_tag_id: id).first
     end
 
     def create_hyper_tag(project)
-      AccessTag.create(project_id: project.id, name: hyper_tag_name, hyper_tag_id: id, hyper_tag_table: self.class.table_name)
+      AccessTag.create(project_id: project.id, name: hyper_tag_name(project), hyper_tag_id: id, hyper_tag_table: self.class.table_name)
```

Comment:
This will raise NoMethodError, right? Did I miss something?

Reply:
If you don't override, yes it will. If a model includes `Authorization::HyperTagMethods` module, it has to override `hyper_tag_name` method. If not, it will raise `NoMethodError` as safeguard

https://github.com/ubicloud/ubicloud/blob/a4f427c8d0d8a959d55dc9db42ecb992b0659d24/model/vm.rb#L22